src/fable/Fable.Compiler/Replacements.fs

```diff
@@ -1226,7 +1226,11 @@ module private AstPass =
         | "removeAt" ->
             icall "splice" (c.Value, [args.Head; makeConst 1]) |> Some
         | "reverse" when kind = Array ->
-            icall "reverse" (instanceArgs c i.args) |> Some
+            match i.returnType with
+            | Fable.Array _ ->
+                emit i "$0.slice().reverse()" i.args |> Some
```

Comment:
We're already checking this is an array in the guard above (`when kind = Array`) so it's probably not necessary to check "returnType", we just use `emit i "$0.slice().reverse()" i.args |> Some` directly :+1: